src/init/power.rs

```diff
@@ -114,13 +115,14 @@ pub fn spawn_thread_power_button_listener(power_btn_device_path: &str) {
                     }
                 }
                 Err(e) => {
-                    error!(
-                        "Could not parse event from {}: {}",
+                    return Err::<(), anyhow::Error>(anyhow!(
```

Comment:
This `Err` isn't returned to anywhere as far as I know (you don't use the returned `JoinHandle<T>`). The thread is spawned off and the function continues. The use of the explicit type is a warning flag for me that the code is not going to work as you are intending.

Reply:
Oh yes. Indeed, this would silently fail because the caller is ignoring the return result of `spawn_thread_power_button_listener`.

My suggestion https://github.com/aurae-runtime/auraed/pull/38/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R197-R207:
* for now, just log if the power button listener started successfully or failed
* todo(not this PR): handle if this thread failed to start.
* todo(not this PR): handle if this thread crashed